### PR TITLE
Datacollection site

### PR DESCRIPTION
**Pull request by guyyoo**

### CODE REVIEW

File: src/rootPages/Designer/ui_work_datacollection_workspace_properties.js [hunk 1]
```diff
@@ -0,0 +1,1036 @@
...
+         this.PopupSortFieldComponent.show($button, null, {
+            pos: "top",
+         });
+      }
+
+      onFilterChange() {
+         const datacollection = this.CurrentDatacollection;
+         if (!datacollection) return;
+
+         const filterValues = this.FilterComponent.getValue();
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/ui_work_datacollection_workspace_properties.js:931]: If we change `this.FilterComponent` to use the `filtercomplexNew()` like I mentioned earlier, that now has a `.isConditionComplete()` method that does this rule check itself.

File: src/rootPages/Designer/ui_work_datacollection_workspace_properties.js [hunk 2]
```diff
@@ -0,0 +1,1036 @@
...
+
+         let allCompconste = true;
+         filterValues.rules.forEach((f) => {
+            // if all 3 fields are present, we are good.
+            if (
+               f.key &&
+               f.rule &&
+               (f.value ||
+                  // these rules do not have input value
+                  f.rule == "is_current_user" ||
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/ui_work_datacollection_workspace_properties.js:944]: it looks like the `FilterComplex.isConditionComplete()` method isn't as complete as this check here.  Go ahead and include these checks into there as well.

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
A few more changes

File: src/rootPages/Designer/ui_work_datacollection_workspace_properties.js [hunk 3]
```diff
@@ -0,0 +1,1036 @@
...
+               };
+            }) || [];
+
+         // Add a current user option to allow select first row that match the current user
+         if (datasource) {
+            const userFields = datasource.fields((f) => f.key == "user");
+            if (userFields.length > 0)
+               dataItems.unshift({
+                  id: "_CurrentUser",
+                  value: L("[Current User]"),
```
[the remark this answers is not in the document]
Comment (guyyoo) [reply src/rootPages/Designer/ui_work_datacollection_workspace_properties.js:834]: Ok, I will bring this up on next standup.

File: src/rootPages/Designer/ui_work_datacollection_workspace_properties.js [hunk 4]
```diff
@@ -0,0 +1,981 @@
...
+
+         this.AB = AB;
+         // {ABFactory}
+
+         /*
+          * _templateListItem
+          *
+          * The Object Row template definition.
+          */
+         this._templateListItem = [
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/ui_work_datacollection_workspace_properties.js:46]: let's change this sort of string building into the newer : 
```
this._templateListItem = `
   <div class='ab-page-list-item'>
     {common.icon()} <span ... > #label# #hasDataCollection#
   </div>
`
```

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
Hey, did you move those additional `isConditionComplete()` conditions into the `FilterComplex` object?

If so, then make this simple string building change and we are good to go.

**Pull request merged**